### ISSUE TITLE
dropbox: ignore and warn about files you can't upload with dropbox

### ISSUE THREAD

**Issue by ncw** [rclone/rclone#108]
If I try uploading a file called `.DS_Store` through the website I get "Sorry! This file name is not allowed.".  

A bit of searching and this is [the best list I can find of file names which aren't allowed](https://www.dropbox.com/en/help/145) - see Ignored files.  I've also seen people saying to avoid windows tempfile names.

Factored from #104

**Comment by philiptzou** [comment]
dropbox still stop working when met ignored files:

```
2015/09/01 21:20:28 .DS_Store: File name disallowed - not uploading
2015/09/01 21:20:28 .DS_Store: Corrupted on transfer: sizes differ 6148 vs 0
2015/09/01 21:20:28 Dropbox root 'shared': Not deleting files as there were IO errors
```

I think you need to re-open this issue.

**Comment by ncw** [comment]
@philiptzou I have verified this - thanks for the report!

I'll fix shortly

```
/tmp$ mkdir /tmp/dbtest
/tmp$ cd /tmp/dbtest/
/tmp/dbtest$ echo hello > floop.txt
/tmp/dbtest$ echo hello > .DS_Store
/tmp/dbtest$ rclone -v sync /tmp/dbtest dropbox:dbtest
[snip]
2015/09/02 08:35:42 .DS_Store: File name disallowed - not uploading
2015/09/02 08:35:42 .DS_Store: Corrupted on transfer: sizes differ 6 vs 0
2015/09/02 08:35:42 .DS_Store: Removing failed copy
2015/09/02 08:35:42 .DS_Store: Failed to remove failed copy: Path '/dbtest/.ds_store' not found
```

I think that Update probably needs to return a sentinel error which can be detected in Copy/Sync.

**Comment by philiptzou** [comment]
@ncw 

Great! I dived into the code some bit and I found there are some other places that may also need this "ignored files" function. For example, the [`fs.Check`](https://github.com/ncw/rclone/blob/master/fs/operations.go#L501) function currently doesn't handle the ignored files on local side so the comparison always shows the changes of `.DS_Store` and others.

Furthermore, I noticed there will be [some other rsync features](https://github.com/ncw/rclone/blob/master/notes.txt#L42) be implemented in future. I perceive it is possible to implement a common file exclusion function.
